test/IECore/MurmurHashTest.py

```diff
@@ -232,6 +232,48 @@ def testDataAccess( self ) :
 			h2 = IECore.MurmurHash( h.h1(), h.h2() )
 			self.assertEqual( h, h2 )
 
+	def testBoolVector( self ) :
+
+		# Bool vectors are a bit tricky since we have to repack the bits in order to hash them,
+		# so lets do some more exhaustive testing
+		hashes = set()
+
+		for i in range( 300 ):
+			h1 = IECore.MurmurHash()
+			h1.append( IECore.BoolVectorData( [0]*i ) )
+			h2 = IECore.MurmurHash()
+			h2.append( IECore.BoolVectorData( [1]*i ) )
+			self.assertFalse( h1 in hashes )
+			self.assertFalse( h2 in hashes )
+			hashes.add( h1 )
+			hashes.add( h2 )
+			check = IECore.MurmurHash()
+			check.append( IECore.BoolVectorData( [0]*i ) )
+			self.assertEqual( check, h1 )
+			check = IECore.MurmurHash()
+			check.append( IECore.BoolVectorData( [1]*i ) )
+			self.assertEqual( check, h2 )
```

Comment:
Unnerving indeed.  I've added some more tests, squashed in since the commit is simple.  For a given length, I now test:
1) All zeros
2) All ones
3) All zeros except last is 1
4) All ones exept last is 0
5) Ones up to length, padded out to fixed size with zeros

The last 3 are new, and all 3 would catch this particular error.  Hopefully I haven't made any less obvious mistakes.